platform/tests/TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -510,6 +511,31 @@ static control_t test_printf_x(const size_t call_count)
     result_baseline = make_test_string("x: ", 11259375, BASE_16, "\r\n");
     TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
 
+    // Test prepending zeros and width size
+    result_minimal = mbed_printf("04X: %04X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "04X: %04X\r\n", 0x04F6);
+    result_baseline = sizeof("04X: 04F6\r\n") - 1;
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_file);
+
+    result_minimal = mbed_printf("0X: %0X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "0X: %0X\r\n", 0x04F6);
+    result_baseline = sizeof("0X: 4F6\r\n") - 1;
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_minimal);
+    TEST_ASSERT_EQUAL_INT(result_baseline, result_file);
+
+    result_minimal = mbed_printf("6X: %6X\r\n", 0x04F6);
+    result_file = mbed_fprintf(stderr, "6X: %6X\r\n", 0x04F6);
+    result_baseline = sizeof("6X: 4F6\r\n") - 1;
```

Comment:
This is not the expected output. The string should have a spaces prepended.

Reply:
Space at which position?